src/site/content/en/angular/route-level-code-splitting/index.md

```diff
@@ -0,0 +1,202 @@
+  declarations: [NyanComponent],
+  imports: [
+    RouterModule.forChild([{
+      path: '',
+      pathMatch: 'full',
+      component: NyanComponent
+    }])
+  ]
+})
+export class NyanModule {}
+```
+
+This code renders `NyanComponent` when the user navigates to [https://example.com/nyan](https://example.com/nyan).
+
+To check that the Angular router downloads the `nyan.module` lazily in your local environment:
+
+{% Instruction 'devtools-network', 'ol' %}
+
+1. Click **NYAN** in the sample app.
+1. Note that the `nyan-nyan-module.js` file appears in the network tab.
+
+![Lazy-loading of JavaScript bundles with route-level code-splitting](code-splitting-network-tab.png "Lazy-loading of JavaScript bundles with route-level code-splitting")
+
+_Find this example [on GitHub](https://github.com/mgechev/code-splitting-web-dev/tree/lazy/src)._
+
+### Show a spinner
+
+Right now, when the user clicks the **NYAN** button, the application doesn't indicate that it's loading JavaScript in the background. To give the user feedback while loading the script you'll probably want to add a spinner.
+
+To do that, start by adding markup for the indicator inside the `router-outlet` element in `app.component.html`:
+
+```html
+<router-outlet>
+  <span class="loader" *ngIf="loading"></span>
+</router-outlet>
+```
+
+Then add an `AppComponent` class to handle routing events. This class will set the `loading` flag to `true` when it hears the `RouteConfigLoadStart` event and set the flag to `false` when it hears the `RouteConfigLoadEnd` event.
+
+```javascript
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  loading: boolean;
+  constructor(router: Router) {
+    this.loading = false;
+    router.events.subscribe(
+      (event: RouterEvent): void => {
+        if (event instanceof NavigationStart) {
+          this.loading = true;
+        } else if (event instanceof NavigationEnd) {
+          this.loading = false;
+        }
+      }
+    );
+  }
+}
+```
+
+In the example below we've introduced artificial 500ms latency so that you can see the spinner in action.
+
+<div class="glitch-embed-wrap" style="height: 480px; width: 100%;">
+  <iframe
+    src="https://stackblitz.com/github/mgechev/code-splitting-web-dev/tree/lazy?embed=1&file=src/app/app.component.ts&view=preview"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+In this post we optimized the app's initial load time at the cost of slowing down navigation to pages. In the [next post](/route-preloading) on route preloading we'll see how we can workaround this problem!
+
+## Conclusion
+
+You can shrink the bundle size of your Angular applications by applying route-level code-splitting:
+
+1. Introduce route-level code-splitting in an Angular application.
+1. Use the `ng g module nyan --module app --route nyan` generator to automatically scaffold a dynamically loaded route
```

Comment:
Can we generalize this for readers who aren't trying to create a nyan cat app? :)